server/src/main/java/org/opensearch/transport/InboundHandler.java

```diff
@@ -322,10 +342,25 @@ private static void sendErrorResponse(String actionName, TransportChannel transp
         }
     }
 
-    private void handlerResponseError(StreamInput stream, final TransportResponseHandler<?> handler) {
+    private void handlerResponseError(final long requestId, StreamInput stream, final TransportResponseHandler<?> handler) {
         Exception error;
         try {
             error = stream.readException();
+
+            // Check the entire message has been read
+            final int nextByte = stream.read();
+            // calling read() is useful to make sure the message is fully read, even if there is an EOS marker
+            if (nextByte != -1) {
```

Comment:
Treat unconsumed stream as (de)serialization failure and handle it as `TransportSerializationException`

Reply:
I think there are redundant pre-existing code blocks for 
1.  handlerResponseError
2. handleResponse
Can we simply this further

Reply:
Those two are slightly different (fe logging, remote address setting), I am not sure the simplification would make sense here (unless we alter the behaviour slightly)